src/lib/frontend/d_cnf.ml

```diff
@@ -101,20 +101,20 @@ module Cache = struct
   let find_ty id =
     HT.find ae_ty_ht (Id id)
 
-  let fresh_ty ?(is_var = true) ?id () =
+  let fresh_ty ?(is_var = true) ?(name = None) () =
     if is_var
     then Ty.fresh_tvar ()
     else
-      match id with
-      | Some id -> Ty.text [] (Uid.of_dolmen id)
+      match name with
+      | Some n -> Ty.text [] (Uid.of_string n)
       | None -> Ty.fresh_empty_text ()
 
-  let update_ty_store ?(is_var = true) id =
-    let ty = fresh_ty ~is_var ~id () in
+  let update_ty_store ?(is_var = true) ?name id =
+    let ty = fresh_ty ~is_var ~name () in
     store_ty id ty
 
-  let update_ty_store_ret ?(is_var = true) id =
-    let ty = fresh_ty ~is_var ~id () in
+  let update_ty_store_ret ?(is_var = true) ?name id =
+    let ty = fresh_ty ~is_var ~name () in
```

Comment:
Using a polymorphic GADT for `Uid.t` almost works every but I ran into an annoying issue here. 
It makes sense to use `ty_cst` for the constructor for the application of an abstract type in `Ty.t` but in `elim_toplevel_forall` in `D_cnf`, we remove the top level binders for both type and term variables. Basically, we replace a type variable `ty_var` by an uninterpreted type using `Text`. So we need to convert a `ty_var` into a `ty_cst`, which is not permit by `Dolmen`. 

I rollback my modifications done in #1098.